GenericNode.ts

```diff
@@ -0,0 +1,71 @@
+export class GenericNode {
+
+    /**
+     * Member which signifies what type of node this is.
+     */
+    nodeType : GenericNode; 
```

Comment:
This isn't a sum type (tho it could be if you replaced children: GenericNode[] with (AtomNode | CutNode)[]), if you need to check what the children's types are from an OOP standpoint you should use the idiomatic solution with the [`insteanceOf` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof)